apps/aecore/test/aecore_sync_SUITE.erl

```diff
@@ -0,0 +1,460 @@
+-module(aecore_sync_SUITE).
+
+%% common_test exports
+-export(
+   [
+    all/0, groups/0, suite/0,
+    init_per_suite/1, end_per_suite/1,
+    init_per_group/2, end_per_group/2,
+    init_per_testcase/2, end_per_testcase/2
+   ]).
+
+%% test case exports
+-export(
+   [
+    start_first_node/1,
+    start_second_node/1,
+    start_third_node/1,
+    mine_on_first/1,
+    mine_on_second/1,
+    mine_on_third/1,
+    mine_again_on_first/1,
+    restart_first/1,
+    restart_second/1,
+    restart_third/1
+   ]).
+
+-export([proxy/0]).
+
+-include_lib("common_test/include/ct.hrl").
+
+all() ->
+    [
+     {group, two_nodes},
+     {group, three_nodes}
+    ].
+
+groups() ->
+    [
+     {two_nodes, [],
+      [start_first_node,
+       mine_on_first,
+       start_second_node,
+       mine_again_on_first,
+       mine_on_second,
+       restart_second,
+       restart_first]},
+     {three_nodes, [],
+      [start_first_node,
+       mine_on_first,
+       start_second_node,
+       start_third_node,
+       mine_again_on_first,
+       mine_on_second,
+       mine_on_third,
+       restart_third]}
+    ].
```

Comment:
An interesting additional test for future iteration would be:
* Start node 1, mine one block, stop node 1;
* Start node 2, mine two blocks;
* Start node 1;
* Check node 1 syncs to node 2.